pkg/policy/mapstate.go

```diff
@@ -1127,8 +1133,8 @@ func (ms *mapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapState
 				ms.addDependentOnEntry(dep.Key, dep.Value, update.Key, changes)
 			}
 		}
-		ms.authPreferredInsert(newKey, newEntry, features, changes)
 	}
+	ms.authPreferredInsert(newKey, newEntry, features, changes)
```

Comment:
We should move this back. I get that this works because of the `return` statement in the `if newEntry.IsDeny`, but it doesn't make sense semantically here.

Reply:
The relevant change from the commit "policy: Skip checking for deny entries when there are none":
```
@@ -1050,7 +1054,9 @@ func (ms *mapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapState
 			newEntry.AddDependent(update.Key)
 		}
 		ms.addKeyWithChanges(newKey, newEntry, changes)
-	} else {
+		return
+	}
+	if !ms.denies.Empty() {
 		// NOTE: We do not delete redundant allow entries.
 		var dependents []MapChange
 
@@ -1127,8 +1133,8 @@ func (ms *mapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapState
 				ms.addDependentOnEntry(dep.Key, dep.Value, update.Key, changes)
 			}
 		}
-		ms.authPreferredInsert(newKey, newEntry, features, changes)
 	}
+	ms.authPreferredInsert(newKey, newEntry, features, changes)
 }
 
 // IsSuperSetOf checks if the receiver Key is a superset of the argument Key, and returns a
```
From the above it can be seen that `ms.authPreferredInsert` call has been moved out of the former `else` block since now that block is only enteted when there are some deny entries to consider. If there are none we still need to call `ms.authPreferredInsert`.